### PR TITLE
README: add caveat (prepared queries)

### PR DESCRIPTION
**Pull request by yhuelf**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -171,6 +171,15 @@ And this time, the specific configuration is applied.
 Caveats
 --------
 
+`pg_query_settings` doesn't work well with **prepared queries**. More specifically:
+for a given parameter `P`, if the generic plan is selected, and if the executor
+needs to fetch the value of `P`, then it won't get the value specified in the
+`pgqs_config` table. For example, the executor fetches the value of `work_mem`
```
Comment (gleu) [inline README.md:177]: I don't understand the example. It seems a bit complex.

Comment (yhuelf) [reply README.md:177]: What about this?

For example, the executor fetches the value of `work_mem` in order to decide whether to perform an in-memory sort, but at this point the value would have been reset to its default (at the end of the `prepare` statement). There's no problem with parameters that only affect planification, such as `max_parallel_workers_per_gather`.

Comment (ioguix) [reply README.md:177]: Why do you need to explain in details what's wrong with planed queries? The first sentence itself is enough.

If you want, you can document in details the technical problem in source code itself.

Comment (gleu) [reply README.md:177]: Looks good to me.

Comment (yhuelf) [reply README.md]: With the last commit?

Comment (ioguix) [reply README.md]: I keep thinking you don't need to add so much details in the README. It just doesn't work well with prepared queries, so prepared queries are buggy and should be considered as not supported yet. No need to explain why.

Comment (yhuelf) [reply README.md]: OK, I agree, I put everything as code comment and only the mininum in README

**Review by ioguix — APPROVED** [review]
good for me.

**Pull request merged**